packages/fuel-indexer-macros/src/indexer.rs

```diff
@@ -244,14 +244,25 @@ fn process_fn_items(
                 })
                 .collect();
             let sig = resolve_fn_selector(&function.name, &params[..]);
+            let fn_name = function.name.clone();
             let selector = u64::from_be_bytes(sig);
             let ty_id = function.output.type_id;
 
-            quote! {
-                #selector => #ty_id,
-            }
+            (
+                quote! {
+                    #selector => #ty_id,
+                },
+                quote! {
+                   #selector => #fn_name.to_string(),
+                },
+            )
         })
-        .collect::<Vec<proc_macro2::TokenStream>>();
+        .collect::<Vec<(proc_macro2::TokenStream, proc_macro2::TokenStream)>>();
```

Comment:
- With the codgen refactor, we now see that all of this codegen stuff just loops ABIgen TypeDefinitions
- Thankfully we can loop as many times as we want since this is all compile time
- With that said, can we split this out?
  - We're combining two distinct `TokenStream`s under a single `abi_selector_tuple_vec`  (btw I don't think you need to add "vec" suffix here since the code shows that it's clearly a `Vec` 😄 )
  - Maybe leave `abi_selectors` as is, and split out the `selector => fn_name` to `fn_name_selectors` (or something similar?) as its own separate loop?

Reply:
Sure, I can do that. I was going for _elegance and speed_, but you're right that it's at compile time anyways and the operation is hardly going to add that much time to the process.